### PR TITLE
Add fresnelc

### LINKED ISSUE #137: RFC: Evaluate C(x) part of Fresnel integral (fresnelc)
## Checklist

> Please ensure the following tasks are completed before filing an issue.

-   [x] Read and understood the [Code of Conduct][code-of-conduct].
-   [x] Searched for existing issues and pull requests.
-   [x] If this is a general question, searched the [FAQ][faq] for an existing answer.
-   [x] If this is a feature request, the issue name begins with `RFC:`.

## Description

Separate fast-path to evaluate *only* C(x) part of Fresnel integral.

## Related Issues

> Does this issue (or feature request) have any related issues?

See #134 for more info.

## Questions

> Any questions for reviewers?

No.

## Other

No.

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

[faq]: https://github.com/stdlib-js/stdlib/blob/develop/FAQ.md

### PR DESCRIPTION
**Pull request by rreusser**
Resolves #137.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing], including the relevant style guides.
-   [x] Read and understand the [Code of Conduct][code-of-conduct].
-   [x] Read and understood the [licensing terms][license].
-   [x] Searched for existing issues and pull requests **before** submitting this pull request.
-   [x] Filed an issue (or an issue already existed) **prior to** submitting this pull request.
-   [x] Rebased onto latest `develop`.
-   [x] Submitted against `develop` branch.

## Description

> What is the purpose of this pull request?

This pull request adds `fresnelc`.

## Related Issues

> Does this pull request have any related issues?

#137, #140, #135.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

Once fully merged, I'm considering submitting a small refactor in which `fresnel` uses `fresnels` and `fresnelc` for separate evaluation of S(x) and C(x) for small x. That will remove some redundant coefficients from `fresnel` at the cost of maybe one extra absolute value evaluation. Otherwise the duplicate coefficients aren't a big deal. I have no strong preference. Honestly I'd hazard a guess that the presence and ease of access of obscure functions greatly greatly outweighs a 20ns difference in evaluation speed.

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

[license]: https://github.com/stdlib-js/stdlib/blob/develop/LICENSE

### CODE REVIEW

**Comment by rreusser** [comment]
I guess what I'm trying to say is that the *only* reason for `fresnels` and `fresnelc` to exist is to avoid the extra line of code it takes to to calculate both S(x) *and* C(x) and throw one away when you just want one of two.

**Comment by kgryte** [comment]
@rreusser Sounds reasonable. Will do the reviews later tonight/tomorrow. You have a preference for review order?

**Comment by rreusser** [comment]
No strong preference. Probably `fresnel`, then `fresnels`, then `fresnelc`. They're *very* similar.

File: lib/node_modules/@stdlib/math/base/special/fresnelc/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# fresnelc
+
+> Compute the C(x) term of the [Fresnel integral][fresnel-integral].
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/fresnelc/README.md]: For consistency with `S(x)`

> Compute the Fresnel integral C(x).

File: lib/node_modules/@stdlib/math/base/special/fresnelc/docs/repl.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
+
+{{alias}}( x )
+    Computes Fresnel integral C(x).
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/fresnelc/docs/repl.txt]: Computes the Fresnel...

File: lib/node_modules/@stdlib/math/base/special/fresnelc/docs/repl.txt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    Parameters
+    ----------
+    x: number
+        Input value.
+
+    Returns
+    -------
+    y: number
+        S(x).
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/fresnelc/docs/repl.txt]: C(x).

File: lib/node_modules/@stdlib/math/base/special/fresnelc/test/fixtures/c/cephes/runner.c [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+* Generates test fixtures.
+*
+* @param x     domain
+* @param len   number of values in the domain
+* @param name  output filename
+*/
+void generate( double *x, const unsigned int len, const char *name ) {
+	unsigned int i;
+	double S;
+	double *C;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/fresnelc/test/fixtures/c/cephes/runner.c]: Move to line above.

**Review by kgryte — CHANGES_REQUESTED** [review]

File: lib/node_modules/@stdlib/math/base/special/fresnelc/README.md (hunk 1 above)
Comment (rreusser) [reply lib/node_modules/@stdlib/math/base/special/fresnelc/README.md]: ✅

File: lib/node_modules/@stdlib/math/base/special/fresnelc/test/fixtures/c/cephes/runner.c (hunk 4 above)
Comment (rreusser) [reply lib/node_modules/@stdlib/math/base/special/fresnelc/test/fixtures/c/cephes/runner.c]: ✅

File: lib/node_modules/@stdlib/math/base/special/fresnelc/docs/repl.txt (hunk 3 above)
Comment (rreusser) [reply lib/node_modules/@stdlib/math/base/special/fresnelc/docs/repl.txt]: Good catch. ✅

**Review by kgryte — APPROVED** [review]

**Comment by kgryte** [comment]
@rreusser Thanks yet again! Will merge once CircleCI finishes. :)

**Pull request merged**